server.js

```diff
@@ -89,16 +98,21 @@ io.on("connection", client => {
   // TODO Create a subscription for this new client to some occupancy changes
   // TODO subscribe to realOccupancyChanges AND simulatedOccupancyChanges
   // TODO subscribe to realOccupancyChanges AND simulatedOccupancyChanges
-  const sub = simulatedOccupancyChanges.subscribe(notifyClient);
+  const sub = merge(
+    simulatedOccupancyChanges,
+    roomHoldOccupancyChanges
+  ).subscribe(notifyClient);
 
   client.on("disconnect", () => {
     sub.unsubscribe();
     console.log("Client disconnected");
   });
 
-  // TODO "holdRoom" types of client actions are ones we went to process
+  // TODO Process holdRoom actions through the store so new clients
+  // will get the actual state. Later, we'll persist the change in the db
   client.on("holdRoom", ({ num, hold }) => {
     console.log("Recv: holdRoom, " + JSON.stringify({ num, hold }));
+    agent.process({ type: "holdRoom", payload: { num, hold } });
```

Comment:
Chronologically, this is the first time our server hears of a `holdRoom` action - so we hand it to the `agent` for processing right away. Without doing the Observable `agent.allOfType('holdRoom')` would remain empty!